flying-saucer-core/src/main/java/org/xhtmlrenderer/layout/SharedContext.java

```diff
@@ -65,7 +65,7 @@ public class SharedContext {
 
     private boolean interactive = true;
 
-    private Map idMap;
+    private Map<String, Box> idMap;
```

Comment:
Consider instantiating the `Map` to avoid potential race-condition.

Reply:
done